ff/src/biginteger/mod.rs

```diff
@@ -137,4 +138,18 @@ pub trait BigInteger:
         *self = Self::read(reader)?;
         Ok(())
     }
+
+    /// Takes two slices of u64 representing big integers and returns a bigger
+    /// BigInteger of type Self representing their product. Preferably used
+    /// only for even NUM_LIMBS. We require the invariant that this.len() ==
+    /// other.len() == NUM_LIMBS / 2
+    fn mul_no_reduce(this: &[u64], other: &[u64]) -> Self;
+
+    /// Similar to `mul_no_reduce` but accepts slices of len == NUM_LIMBS and
+    /// only returns lower half of the result
+    fn mul_no_reduce_lo(this: &[u64], other: &[u64]) -> Self;
```

Comment:
Same for this

Reply:
Maybe we can change the parameters here? `fn mul_no_reduce_lo(&self, other: &Self) -> Self`? Does your later code rely heavily on slices?

Another approach would be to use something like `num-bigint` for those places, because that explicitly supports these kinds of operations, and I imagine that the bigint ops won't dominate the GLV ops?